config/v1/types_dns.go

```diff
@@ -26,6 +26,25 @@ type DNSSpec struct {
 	// For example, given the base domain `openshift.example.com`, an API server
 	// DNS record may be created for `cluster-api.openshift.example.com`.
 	BaseDomain string `json:"baseDomain"`
+	// publicZone is the location where all the DNS records that are publicly accessible to
+	// the internet exist.
+	// If this field is nil, no public records should be created.
+	// +optional
+	PublicZone *DNSZone `json:"publicZone,omitempty"`
+	// privateZone is the location where all the DNS records that are only available internally
+	// to the cluster exist.
+	PrivateZone DNSZone `json:"privateZone"`
```

Comment:
Public vs. private can come out of the AWS API (e.g. [here][1]).  Can we just have a slice of zone IDs?

[1]: https://github.com/openshift/installer/blob/v0.12.0/pkg/destroy/aws/aws.go#L341-L342

Reply:
> I think the configuration should be specifying, go create public records here and private records here. not the other way around...

Are you concerned that sometimes this information would not be available from the AWS API?  Or are you concerned about the ingress operator getting the check wrong and accidentally pushing private records to a public zone?  I don't see why bugs on this front would be more likely in ingress code than in installer code.

On the other hand, I don't see a need for multiple public or private zones either, so having separate properties isn't a big change.  If the ingress operator wants to structure it's handler in a loop and we get separate entries here, it can use `[]*DNSZone{&spec.PrivateZone, spec.PublicZone}` with a guard for `nil` inside the loop.

Reply:
This has to be infrastructure independent for sure.

Reply:
Does this need to be a pointer too (or use a zone slice), so we have a way to tell ingress not to worry about DNS in the libvirt/BYOI cases?